util/chplenv/chpl_gpu.py

```diff
@@ -12,31 +12,38 @@ def get():
 
 @memoize
 def get_cuda_path():
-    chpl_cuda_path = os.environ.get("CHPL_CUDA_PATH")
-    if chpl_cuda_path:
-        return chpl_cuda_path
+    if chpl_locale_model.get() == 'gpu':
+        chpl_cuda_path = os.environ.get("CHPL_CUDA_PATH")
+        if chpl_cuda_path:
+            return chpl_cuda_path
 
-    exists, returncode, my_stdout, my_stderr = utils.try_run_command(["which",
-                                                                      "nvcc"])
+        exists, returncode, my_stdout, my_stderr = utils.try_run_command(["which",
+                                                                          "nvcc"])
 
-    if exists and returncode == 0:
-        chpl_cuda_path = "/".join(my_stdout.strip().split("/")[:-2])
-        return chpl_cuda_path
-    else:
-        return ""
+        if exists and returncode == 0:
+            chpl_cuda_path = "/".join(os.path.realpath(my_stdout).strip().split("/")[:-2])
+            return chpl_cuda_path
+        else:
+            error("Can't find cuda")
+
+    return ""
```

Comment:
Is there any particular reason why we're returning the empty string if `CHPL_LOCALE_MODEL != gpu.`

Seems like what we really want is `assert(chpl_locale_model.get() == gpu)`

Reply:
Good question, and it is something I don't have a good answer for. Some connections between `printchplenv` and runtime/compiler Makefiles are a bit confusing to me.

When configuring runtime we run `printchplenv`. We need the CUDA path returned here for building the runtime correctly. So, adding an assert like that and not having it trigger for non-gpu locale models is beyond my understanding of how `printchplenv` works.

Also, to be sure, this is not a different behavior than what we have today. The difference is we error if we can't find CUDA with GPU locale model, as opposed to having an empty path for `CHPL_CUDA_PATH`.